tests/js/server/aql/aql-optimizer-rule-last-access-on-path.js

```diff
@@ -0,0 +1,189 @@
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Michael Hackstein
+// / @author Copyright 2023, ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const gm = require("@arangodb/general-graph");
+const {db} = require("@arangodb");
+const ruleName = "optimize-traversal-last-element-access";
+
+function TraversalOptimizeLastPathAccessTestSuite() {
+    const graphName = "UnitTestGraph";
+    const vc = "UnitTestVertexCollection";
+    const ec = "UnitTestEdgeCollection";
+    const startVertex = "UnitTestVertexCollection/1";
+
+    const injectData = () => {
+        // There is nothing special to this dataset, just something that matches the filters as used in the tests later;
+        // The only thing: We have the startVertex included
+        const vertices = [];
+        for (let i = 0; i < 100; ++i) {
+            vertices.push({_key: `${i}`, name: `${i % 2 === 0 ? "test" : "foo"}`});
+        }
+        db[vc].save(vertices);
+
+        const edges = [];
+        for (let i = 0; i < 100; ++i) {
+            for (let j = 0; j < 100; j += i + 2) {
+                edges.push({_from: `${vc}/${i}`, _to: `${vc}/${j}`, label: `${i % 2 === 0 ? "test" : "foo"}`, timestamp: j});
+            }
+        }
+        db[ec].save(edges);
+    };
+
+    const makePath = (length, keyOffset = 0) => {
+        const vertices = [];
+        const edges = [];
+        for (let i = keyOffset; i < keyOffset + length; ++i) {
+            vertices.push({_key: `${i}`, _id: `v/${i}`});
+            if (i > 0) {
+                edges.push({_key: `${i - 1}-${i}`, _id: `e/${i - 1}-${i}`, _from: `v/${i - 1}`, _to: `v/${i}`});
+            }
+        }
+        return {vertices, edges};
+    };
+    // These paths are used to "simulate" traversal output, to make sure optimizer does not do funny stuff
+    const hardCodedPaths = [
+        makePath(1, 0),
+        makePath(2, 0),
+        makePath(3, 0),
+        makePath(1, 10),
+        makePath(2, 10),
+        makePath(3, 10),
+    ];
+
+    const isReferenceToPath = (refNode) => {
+        const {typeID, name} = refNode;
+        // Only Reference(45) to variable (p) are allowed
+        return typeID === 45 && name === "p";
+    };
+    const isLastElementAccess = (indexedAccess) => {
+        if (indexedAccess.subNodes.length !== 2) {
+            // Index access needs exactly 2 members
+            return false;
+        }
+        const [path, index] = indexedAccess.subNodes;
+        {
+            // Test index
+            const {typeID, value} = index;
+            if (typeID !== 40 || value !== -1) {
```

Comment:
Can we use the type name instead of the typeID? Seeing a typeID value `40` isn't really intuitive when looking at the test code.